### ISSUE TITLE
The control autoscaling_group_launch_configuration_requires_imdsv2 fails if AWS Launch Template is used

### ISSUE THREAD

**Issue by serhii-ciq** [toniblyx/prowler#6047]
### Steps to Reproduce

* Create a Launch Template with the required option imdsv2
* Create an ASG using this template
* Run the Prowler scan
```
prowler aws -c autoscaling_group_launch_configuration_requires_imdsv2
```

### Expected behavior

If I use a Launch Template instead of a Launch Configuration for creating an ASG the check should not fail

### Actual Result with Screenshots or Logs

![image](https://github.com/user-attachments/assets/9fa03402-ac9e-4f81-8d22-ebbd6a2cffe3)

### How did you install Prowler?

From pip package (pip install prowler)

### Environment Resource

Workstation

### OS used

Windows

### Prowler version

Prowler 5.0.0 (You are running the latest version, yay!)

### Pip version

pip 24.2 (python 3.12)

**Comment by MrCloudSec** [comment]
Hi @serhii-ciq, thanks for the heads up! We will review it and let you know when we have a fix for that 😄

**Comment by serhii-ciq** [comment]
Cool, thank you @MrCloudSec

**Comment by danibarranqueroo** [comment]
Hi @serhii-ciq , thanks again for reporting the issue😄. I've opted for the solution explained in the linked PR, where I modified the check to avoid returning findings if there is no launch configuration in the Auto Scaling group. This will prevent incorrect statuses when a launch template is used. Additionally, I’ve created a new check, `ec2_launch_template_imdsv2_required`, to handle the IMDSv2 required configuration for launch templates. The PR is already open, and once it's reviewed and approved, you’ll be able to test from master to ensure it works as expected.
